lib/bolt/node/ssh.rb

```diff
@@ -6,13 +6,24 @@
 module Bolt
   class SSH < Node
     def connect
-      options = { logger: @transport_logger,
-                  non_interactive: true }
+      options = {
+        logger: @transport_logger,
+        non_interactive: true
+      }
+
       options[:port] = @port if @port
       options[:password] = @password if @password
+      options[:verify_host_key] = if @insecure
+                                    Net::SSH::Verifiers::Lenient.new
+                                  else
+                                    Net::SSH::Verifiers::Secure.new
+                                  end
```

Comment:
This overrides whatever is in `~/.ssh/config` which I don't think is desirable, e.g. if it contains `StrictHostKeyChecking no` for the host/domain you're trying to connect to.

Reply:
`net-ssh` doesn't parse the `StrictHostKeyChecking` value for a host, e.g. given:

```
  StrictHostKeyChecking no
```

net-ssh doesn't load the value:

```
Net::SSH::Config.for(@host)
{:auth_methods=>["none", "publickey", "password", "keyboard-interactive"], :user=>"root", :keys=>["~/.ssh/id_rsa", "~/.ssh/josh_puppetlabs"], :send_env=>[/^LANG$/, /^LC_.*$/]}
```